Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/types.go

```diff
@@ -116,3 +116,17 @@ type Unknown struct {
 }
 
 func (*Unknown) IsAnAPIObject() {}
+
+// Unstructured allows objects that do not have Golang structs registered to be manipulated
+// generically. This can be used to deal with the API objects from a plug-in. Unstructured
+// objects still have functioning TypeMeta features-- kind, version, etc.
+// TODO: Make this object have easy access to field based accessors and settors for
+// metadata and field mutatation.
+type Unstructured struct {
```

Comment:
json.Decoder has a token scanner that it uses to identify fields.  You perform O(N) less object creation.  Decode time and memory use is linear with number of fields in your destination structure.

----- Original Message -----
> > @@ -116,3 +116,17 @@ type Unknown struct {
> >  }
> >  
> >  func (*Unknown) IsAnAPIObject() {}
> > +
> > +// Unstructured allows objects that do not have Golang structs registered
> > to be manipulated
> > +// generically. This can be used to deal with the API objects from a
> > plug-in. Unstructured
> > +// objects still have functioning TypeMeta features-- kind, version, etc.
> > +// TODO: Make this object have easy access to field based accessors and
> > settors for
> > +// metadata and field mutatation.
> > +type Unstructured struct {
> 
> > Unknown is much less expensive
> 
> That's not obvious to me.  Don't you have to parse the json anyway,
> effectively creating the map, until you find hits for the `kind` and
> `version` regardless?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1959/files#r29335924